packages/server/src/index.ts

```diff
@@ -13,15 +13,22 @@ import { authPlugin } from './plugins/auth-plugin.js';
 async function main() {
   const application = await createGraphQLApp(env);
 
+  // While createGraphQLApp initializes the application, we use buildSubgraphSchema to build the Subgraph Schema.
   const yoga = createYoga({
+    schema: application.schema,
+    graphqlEndpoint: '/legacy',
+    graphiql: {
+      title: 'Legacy Subgraph',
+    },
+    landingPage: true,
     plugins: [
       authPlugin(),
```

Comment:
The entier authorization process should be done in the gateway, and not in the subgraph. 

@gilgardosh some questions:
1. what kind of token do you use? who's the issuer and what's the format? 
2. can we move the token validation to be at the gateway level? or this is something we want to do later? 
3. what kind of data is extracted from the token? I'm asking because if the GW will be in charge of validating the token, it will also be able to pass some token payload (in case of JWT) to the subgraph.

Reply:
@dotansimha the current system is for general role authentication. The auth token is basic hash of `{name, pass}` object, Where name actually refers a role.
The server plugin validates the name(role) and password against the info on `AUTHORIZED_USERS` env var.

1. Passwords are encrypted using [bcrypt](https://www.npmjs.com/package/bcrypt)
2. Yes, we can move authentication to the gateway level, but we should then forward the role and derived admin business ID to the subschemas. A thing to keep in mind is that by that we leave the subschemas unprotected if someone reaches them directly.
3. answered above

Let me know if it's still unclear